thorin/check.h

```diff
@@ -38,41 +38,33 @@ class Infer : public Def {
     flags_t& rank() { return flags_; }
 
     THORIN_DEF_MIXIN(Infer)
-    friend class Checker;
+    friend class Check;
 };
 
-class Checker {
+class Check {
 public:
-    Checker(World& world)
-        : world_(&world) {}
-
-    World& world() const { return *world_; }
-
-    /// Are @p d1 and @p d2 α-equivalent?
-    bool equiv(Ref d1, Ref d2);
+    /// Are d1 and d2 α-equivalent?
+    /// * In @p infer mode, type inference is happening and Infer%s will be resolved, if possible.
+    ///     Also, two *free* but *different* Var%s **are** considered α-equivalent.
+    /// * When **not* in @p infer mode, no type inference is happening and Infer%s will not be touched.
+    ///     Also, Two *free* but *different* Var%s are **not** considered α-equivalent.
+    template<bool infer = true> static bool alpha(Ref d1, Ref d2) { return Check().alpha_<infer>(d1, d2); }
```

Comment:
So we loose basically all caching across two `alpha` calls?

Reply:
yes, but I always found the caching questionable to begin with as we could mutate the alpha-equivalence away. We can later on still try to add some caching mechanism if this is too slow.